pkg/models/tenant/tenant.go

```diff
@@ -407,6 +414,95 @@ func (t *tenantOperator) Events(user user.Info, queryParam *eventsv1alpha1.Query
 	})
 }
 
+func (t *tenantOperator) QueryLogs(user user.Info, query *loggingv1alpha2.Query) (*loggingv1alpha2.APIResponse, error) {
+	iNamespaces, err := t.listIntersectedNamespaces(user,
+		stringutils.Split(query.WorkspaceFilter, ","),
+		stringutils.Split(query.WorkspaceSearch, ","),
+		stringutils.Split(query.NamespaceFilter, ","),
+		stringutils.Split(query.NamespaceSearch, ","))
+	if err != nil {
+		klog.Error(err)
+		return nil, err
+	}
+
+	namespaceCreateTimeMap := make(map[string]time.Time)
+	for _, ns := range iNamespaces {
+		namespaceCreateTimeMap[ns.Name] = ns.CreationTimestamp.Time
+	}
+
+	sf := loggingclient.SearchFilter{
+		NamespaceFilter: namespaceCreateTimeMap,
+		WorkloadSearch:  stringutils.Split(query.WorkloadSearch, ","),
+		WorkloadFilter:  stringutils.Split(query.WorkloadFilter, ","),
+		PodSearch:       stringutils.Split(query.PodSearch, ","),
+		PodFilter:       stringutils.Split(query.PodFilter, ","),
+		ContainerSearch: stringutils.Split(query.ContainerSearch, ","),
+		ContainerFilter: stringutils.Split(query.ContainerFilter, ","),
+		LogSearch:       stringutils.Split(query.LogSearch, ","),
+		Starttime:       query.StartTime,
+		Endtime:         query.EndTime,
+	}
```

Comment:
Need to filter out the namespace（or worksapce） which the user doesn't have rights to list logs.
https://github.com/kubesphere/kubesphere/blob/master/pkg/models/tenant/tenant.go#L364-L382